docs/user-guide/configure.md

```diff
@@ -2,14 +2,46 @@
 
 Stylelint expects a configuration object, and looks for one in a:
 
+- `stylelint.config.js` or `.stylelintrc.js` file
+  - Which module system to use depends on your [default module system configuration](https://nodejs.org/api/packages.html#determining-module-system) for Node.js (e.g., `"type": "module"` in `package.json`).
+- `stylelint.config.mjs` or `.stylelintrc.mjs` file using `export default` (ES module)
+- `stylelint.config.cjs` or `.stylelintrc.cjs` file using `module.exports` (CommonJS)
+- `.stylelintrc.json`, `.stylelintrc.yml`, or `.stylelintrc.yaml` file
+- `.stylelintrc` file in JSON or YAML format
+  - We recommend adding an extension (e.g., `.json`) to help your editor provide syntax checking and highlighting.
 - `stylelint` property in `package.json`
-- `.stylelintrc` file
-- `.stylelintrc.{cjs,js,json,yaml,yml}` file
-- `stylelint.config.{cjs,mjs,js}` file
 
-Starting from the current working directory, Stylelint stops searching when one of these is found. Alternatively, you can use the [`--config` or `configFile` option](options.md#configfile) to short-circuit the search.
+ES module example:
+
+```js
+export default {
+  rules: {
+    "block-no-empty": true
+  }
+};
+```
+
+CommonJS example:
+
+```js
+module.exports = {
+  rules: {
+    "block-no-empty": true
+  }
+};
+```
 
-The `.stylelintrc` file (without extension) can be in JSON or YAML format. We recommend adding an extension to help your editor provide syntax checking and highlighting.
+JSON example:
+
+```json
+{
+  "rules": {
+    "block-no-empty": true
+  }
+}
+```
```

Comment:
[ask] Are these examples redundant? Feel free to give me feedback.

Reply:
Sample code is so important, it can clearly express how users need to migrate.

In addition to rules, can you add additional examples such as `extends`, `plugins`, `overrides`, `ignoreFiles`?
Or are these the same APIs as before?

Reply:
Thanks for the feedback.

> In addition to rules, can you add additional examples such as `extends`, `plugins`, `overrides`, `ignoreFiles`?
> Or are these the same APIs as before?

Yes, the same as before. I think keeping these examples simple in the top section may be good since each section already has a more detailed example.

Reply:
OK, I understand. Thank you for your reply.